pkg/controller/handlers/toolreference/toolreference.go

```diff
@@ -447,27 +479,20 @@ func isCommandPreferred(existing, newer string) bool {
 }
 
 type catalogEntryInfo struct {
-	ID              int    `json:"id"`
-	Path            string `json:"path"`
-	DisplayName     string `json:"displayName"`
-	FullName        string `json:"fullName"`
-	URL             string `json:"url"`
-	Description     string `json:"description"`
-	Stars           int    `json:"stars"`
-	ReadmeContent   string `json:"readmeContent"`
-	Language        string `json:"language"`
-	Metadata        string `json:"metadata"`
-	License         string `json:"license"`
-	Icon            string `json:"icon"`
-	Manifest        string `json:"manifest"`
-	ToolDefinitions string `json:"toolDefinitions"`
-}
-
-type mcpServerManifest struct {
-	Name        string            `json:"name"`
-	Description string            `json:"description"`
-	Category    string            `json:"category"`
```

Comment:
The `Name`, `Description`, and `Category` fields were completely unused, so I removed this type and we just use `mcpServerConfig` directly as the manifest type.